lib/api/2.0/ibms.js

```diff
@@ -0,0 +1,118 @@
+// Copyright 2016, EMC, Inc.
+
+'use strict';
+
+var injector = require('../../../index.js').injector;
+var controller = injector.get('Http.Services.Swagger').controller;
+var schemaApiService = injector.get('Http.Api.Services.Schema');
+var nameSpace = '/api/2.0/ibms/definitions/';
+var waterline = injector.get('Services.Waterline');
+var eventsProtocol = injector.get('Protocol.Events');
+var Errors = injector.get('Errors');
+var _ = injector.get('_'); // jshint ignore:line
+
+
+// GET /api/2.0/ibms
+var ibmsGet = controller(function(req) {
+    return waterline.ibms.find(req.query);
+});
+
+// PUT /api/2.0/ibms
+var ibmsPut = controller({success: 201}, function(req) {
+    var nodeId = req.swagger.params.body.value.nodeId;
+    delete req.swagger.params.body.value.nodeId;
+    return waterline.ibms.upsertByNode(nodeId, req.swagger.params.body.value);
+});
+
+// GET /api/2.0/ibms/identifier
+var ibmsGetById = controller(function(req) {
+    return waterline.ibms.needByIdentifier(req.swagger.params.identifier.value);
+});
+
+// PATCH /api/2.0/ibms/identifier
+var ibmsPatchById = controller( function(req) {
+    var ibmId = req.swagger.params.identifier.value;
+    var values = req.swagger.params.body.value;
+    return waterline.ibms.needByIdentifier(ibmId)
```

Comment:
some of the codes are copied from /lib/services/obm-api-services.js   the code logic is the same. I think don't need another abstract layer to service api, if ibm and obm really almost the similar in document operation. maybe just a map/dict could reduce the duplication.  maintain obm/ibm in /lib/services/bm-api-service.js or lib/services/nm-api-service.js(network management)  all the document openration like waterline.obms.xxxx could be waterline.nm[type].xxx, just a method I could think of to de-duplication. you decide use it or not.